src/pages/iou/request/step/IOURequestStepDate.tsx

```diff
@@ -37,27 +44,39 @@ type IOURequestStepDateOnyxProps = {
 };
 
 type IOURequestStepDateProps = IOURequestStepDateOnyxProps &
-    WithWritableReportOrNotFoundProps<typeof SCREENS.MONEY_REQUEST.STEP_WAYPOINT> & {
+    WithWritableReportOrNotFoundProps<typeof SCREENS.MONEY_REQUEST.STEP_DATE> & {
         /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
         transaction: OnyxEntry<OnyxTypes.Transaction>;
+
+        /** The report linked to the transaction */
+        report: OnyxEntry<Report>;
     };
 
 function IOURequestStepDate({
     route: {
-        params: {action, iouType, reportID, backTo},
+        params: {action, iouType, reportID, backTo, reportActionID},
     },
     transaction,
     splitDraftTransaction,
     policy,
     policyTags,
     policyCategories,
+    reportActions,
+    report,
+    session,
 }: IOURequestStepDateProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const isEditing = action === CONST.IOU.ACTION.EDIT;
     // In the split flow, when editing we use SPLIT_TRANSACTION_DRAFT to save draft value
     const isEditingSplitBill = iouType === CONST.IOU.TYPE.SPLIT && isEditing;
     const currentCreated = isEditingSplitBill && !lodashIsEmpty(splitDraftTransaction) ? TransactionUtils.getCreated(splitDraftTransaction) : TransactionUtils.getCreated(transaction);
+    const parentReportAction = reportActions?.[(isEditingSplitBill ? reportActionID : report?.parentReportActionID) ?? 0];
+    const canEditingSplitBill =
+        isEditingSplitBill && session && parentReportAction && session.accountID === parentReportAction.actorAccountID && TransactionUtils.areRequiredFieldsEmpty(transaction);
+    const canEditMoneyRequest = isEditing && ReportUtils.canEditFieldOfMoneyRequest(parentReportAction ?? null, CONST.EDIT_REQUEST_FIELD.DATE);
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    const shouldShowNotFound = !IOUUtils.isValidMoneyRequestType(iouType) || (isEditing && !canEditMoneyRequest && !canEditingSplitBill);
```

Comment:
Why is this change required @FitseTLT? I think this is getting way too complicated.

Reply:
@FitseTLT Isn't `ReportUtils.canEditFieldOfMoneyRequest` function enough to do the job? Can you share the relevant checks on tags and categories page?

Reply:
https://github.com/Expensify/App/blob/63223bcc99a930f85cc7d03f7df6730597936607/src/pages/iou/request/step/IOURequestStepTag.tsx#L70-L84

Reply:
> Isn't ReportUtils.canEditFieldOfMoneyRequest function enough to do the job

Can you please answer above?

Reply:
Nope it is not enough because we have to include the case of split detail page that is what I have included and I have given you a perfect similar example done for tag.